### PR TITLE
tabletserver: queryctl revamp

### PR DESCRIPTION
**Pull request by sougou**
queryctl was initally created because TabletServer could not
export non-rpc functions. Otherwise, the go rpc framework would
complain about those functions.
However, this has changed because we now have rpc-specific adapters.
So, this means that TabletServer can export any function it pleases.
This has made the objects in queryctl redundant.
This change pushes those thin wrappers straight into TabletServer.
Also, I've moved the mocking interface into its own package. This
prevents test code from intermixing with normal code.
There's not much left in queryctl. So, I'm currently debating if it
should remain as a file. I'm leaving it there for now.
There are still a few stray globals left. They're harmless for now. I've removed
the more problematic checkMySQL.

### CODE REVIEW

File: go/vt/tabletserver/mock/controller.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package mock provides mock interfaces for tabletserver.
+package mock
```
Comment (alainjobart) [inline go/vt/tabletserver/mock/controller.go]: This is way too generic of a name, I don't want to repeat the 'proto' subdir thing where we have to rename them on import everywhere... How about tabletservermock? Or tsmock is you want a shorter name...

File: go/vt/tabletserver/mock/controller.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+	// IsHealthy is the return value for IsHealthy
+	IsHealthyError error
+
+	// ReloadSchemaCount counts how many times ReloadSchema was called
+	ReloadSchemaCount int
+}
+
+// NewController returns an implementation of tabletserver.Controller
+// that is entirely fake
```
Comment (alainjobart) [inline go/vt/tabletserver/mock/controller.go]: So is this a fake or a mock? I would say fake.

Then, fakets as the package name?

**Comment by alainjobart** [comment]
LGTM after comments addressed.

File: go/vt/tabletserver/mock/controller.go (hunk 2 above)
Comment (sougou) [inline go/vt/tabletserver/mock/controller.go]: Based on the docs, this is a mock. I'll fix the comments.

File: go/vt/tabletserver/mock/controller.go (hunk 1 above)
Comment (sougou) [inline go/vt/tabletserver/mock/controller.go]: Sounds good.

**Pull request merged**